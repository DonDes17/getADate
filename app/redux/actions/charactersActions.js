import {
  GET_CHARACTERS,
  GET_CHARACTER_DETAILS,
  GET_MORE_PAGES,
} from '../constants';

// Action creator pour récupérer les personnages
export const getCharacters = () => ({ type: GET_CHARACTERS });

// Action creator pour récupérer une page supplémentaire
export const getMorePages = page => ({ type: GET_MORE_PAGES, page });

// Action creator pour récupérer les datails d'un personnage
export const getCharactersDetails = id => ({ type: GET_CHARACTER_DETAILS, id });
