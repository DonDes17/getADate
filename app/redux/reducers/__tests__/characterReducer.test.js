import CharacterReducer from '../characterReducer';
import { getCharacters, getCharactersDetails } from '../../actions/charactersActions';

describe('CharacterReducer', () => {
  // const initialState = {};

  it('nested data on initial fetch', () => {
    // const expected = {
    //   ...initialState,
    //   loading: true
    // }
    // const actual = CharacterReducer(undefined,getCharacters());
    // expect(actual).toEqual(expected);
    expect(CharacterReducer(undefined, getCharacters())).toMatchSnapshot();
  });

  it('nested data on initial detail fetch', () => {
    // const expected = {
    //   ...initialState,
    //   loading: true
    // }
    // const actual = CharacterReducer(undefined,getCharactersDetails());
    // expect(actual).toEqual(expected);
    expect(CharacterReducer(undefined, getCharactersDetails())).toMatchSnapshot();
  });
});
