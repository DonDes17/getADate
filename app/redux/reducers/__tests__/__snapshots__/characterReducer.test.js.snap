// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterReducer nested data on initial detail fetch 1`] = `
Object {
  "loading": true,
}
`;

exports[`CharacterReducer nested data on initial fetch 1`] = `
Object {
  "loading": true,
}
`;
